application/src/main/kotlin/com/studentcenter/weave/application/meetingTeam/service/application/MeetingTeamCreateInvitationLinkApplicationService.kt

```diff
@@ -0,0 +1,61 @@
+package com.studentcenter.weave.application.meetingTeam.service.application
+
+import com.studentcenter.weave.application.common.security.context.getCurrentUserAuthentication
+import com.studentcenter.weave.application.meetingTeam.port.inbound.MeetingTeamCreateInvitationLinkUseCase
+import com.studentcenter.weave.application.meetingTeam.service.domain.MeetingTeamDomainService
+import com.studentcenter.weave.application.meetingTeam.util.MeetingTeamInvitationService
+import com.studentcenter.weave.domain.meetingTeam.entity.MeetingTeam
+import com.studentcenter.weave.domain.meetingTeam.enums.MeetingTeamStatus
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.*
+
+@Service
+class MeetingTeamCreateInvitationLinkApplicationService(
+    private val meetingTeamInvitationService: MeetingTeamInvitationService,
+    private val meetingTeamDomainService: MeetingTeamDomainService,
+) : MeetingTeamCreateInvitationLinkUseCase {
+
+    @Transactional
+    override fun invoke(meetingTeamId: UUID): MeetingTeamCreateInvitationLinkUseCase.Result {
+        val currentUserId = getCurrentUserAuthentication().userId
+
+        val meetingTeam = meetingTeamDomainService.getById(meetingTeamId)
+            .also {
+                validateCurrentUserIsLeader(
+                    meetingTeamId = it.id,
+                    currentUserId = currentUserId,
+                )
+                validateTeamStatusIsWaiting(it)
+                validateTeamVacancyIsNotFull(it)
+            }
+
+        val meetingTeamInvitation = meetingTeamInvitationService.create(meetingTeam.id)
+
+        return MeetingTeamCreateInvitationLinkUseCase.Result(meetingTeamInvitationLink = meetingTeamInvitation.invitationLink)
+    }
+
+    private fun validateCurrentUserIsLeader(
+        meetingTeamId: UUID,
+        currentUserId: UUID,
+    ) {
+        val meetingTeamLeader =
+            meetingTeamDomainService.getLeaderMemberByMeetingTeamId(meetingTeamId)
+
+        require(meetingTeamLeader.userId == currentUserId) {
+            "팀장만 새로운 팀원을 초대할 수 있어요!"
+        }
+    }
+
+    private fun validateTeamStatusIsWaiting(meetingTeam: MeetingTeam) {
+        require(meetingTeam.status == MeetingTeamStatus.WAITING) {
+            "팀의 정원이 이미 가득 차 있어서 새로운 팀원을 초대할 수 없어요!"
+        }
+    }
+
+    private fun validateTeamVacancyIsNotFull(meetingTeam: MeetingTeam) {
+        require(meetingTeamDomainService.findAllMeetingMembersByMeetingTeamId(meetingTeam.id).size < meetingTeam.memberCount) {
```

Comment:
p4) 멤버의 수는 자주 쓸 것 같아서 Count 메서드를 이용하는 것도 좋을 것 같아요!
```suggestion
        require(meetingTeamDomainService.countMeetingMembersByMeetingTeamId(meetingTeam.id) < meetingTeam.memberCount) {
```

Reply:
요거 조회 코드 짤 때 같이 반영하겠습니다!!